snake_state.py

```diff
@@ -164,6 +164,10 @@ def apple_position(self) -> Tuple[int, int]:
     def snake_length(self) -> int:
         return len(self.__snake) + self.__to_grow
 
+    @property
+    def dist_to_apple(self) -> float:
+        return abs(self.__snake[0] - self.apple_position[0]) + abs(self.__snake[1] - self.apple_position[1])
```

Comment:
Use `self.head_position[0]` instead of `self.__snake[0][0]` (essentially exactly what you already did with apple pos, just for head pos as well)